src/core/src/package_managers/UbuntuProClient.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 3.5+
+
+"""This is the Ubuntu Pro Client implementation"""
+
+
+class UbuntuProClient:
+    def __init__(self, env_layer, composite_logger):
+        self.env_layer = env_layer
+        self.composite_logger = composite_logger
+        self.ubuntu_pro_client_install_cmd = 'sudo apt-get install ubuntu-advantage-tools -y'
+
+    def install_or_update_pro(self):
+        """install/update pro(ubuntu-advantage-tools) to the latest version"""
+        try:
+            # Install Ubuntu Pro Client.
+            code, output = self.env_layer.run_command_output(self.ubuntu_pro_client_install_cmd, False, False)
+            if code == 0:
+                self.composite_logger.log_debug("Ubuntu Pro Client installation successful. " + output)
+                return True
+            else:
+                self.composite_logger.log_debug("Ubuntu Pro Client installation failed. " + output)
+                return False
+        except Exception as error:
+            self.composite_logger.log_debug("Exception in installing Ubuntu Pro Client installation " + repr(error))
```

Comment:
Similar question on status blob, does this error need to be added in status?

Reply:
Yes, this is for telemetry. wanted to see if this happens frequently.

Reply:
nit: "Exception in installing Ubuntu Pro Client installation " => ""Exception in installing Ubuntu Pro Client: " 
(installing .. installation is redundant)

Reply:
Also, Rajasi's question was about if the customer needs to know about this failure. Not initially, may be revisited later.

Reply:
Thanks. made the changes to show single log and return once.